traits/trait_types.py

```diff
@@ -3347,7 +3347,19 @@ class NoneTrait(TraitType):
     """ Defines a trait that only accepts the None value
     """
 
-    info_text = "a None type"
+    info_text = "None"
+
+    default_value = None
+
+    default_value_type = DefaultValue.constant
+
+    def __init__(self, **metadata):
+        super(NoneTrait, self).__init__(**metadata)
+        default_value = metadata.pop("default", None)
+        if default_value is not None:
+            raise ValueError("Cannot set default value {} "
+                             "for NoneTrait".format(default_value))
+        self.metadata = metadata.copy()
```

Comment:
This is done in `super().__init__`; and if you do the check before calling `super()` then you shouldn't need it.